src/Elliptic/SubdomainPreconditioners/MinusLaplacian.hpp

```diff
@@ -79,41 +105,65 @@ class MinusLaplacian
   using Base = LinearSolver::Serial::LinearSolver<LinearSolverRegistrars>;
   using StoredSolverType = tmpl::conditional_t<std::is_abstract_v<Solver>,
                                                std::unique_ptr<Solver>, Solver>;
+  // Identifies an external block boundary, for boundary conditions
+  using BoundaryId = std::pair<size_t, Direction<Dim>>;
 
  public:
   static constexpr size_t volume_dim = Dim;
   using options_group = OptionsGroup;
   using poisson_system =
       Poisson::FirstOrderSystem<Dim, Poisson::Geometry::FlatCartesian>;
+  using BoundaryConditionsBase =
+      typename poisson_system::boundary_conditions_base;
   using SubdomainOperator = elliptic::dg::subdomain_operator::SubdomainOperator<
       poisson_system, OptionsGroup, tmpl::list<>,
       tmpl::list<Poisson::BoundaryConditions::Robin<Dim>>>;
   using SubdomainData = ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
       Dim, tmpl::list<Poisson::Tags::Field>>;
+  // Associates Dirichlet or Neumann conditions to every external block
+  // boundary. For every configuration of this type we'll need a separate
+  // solver, which may cache the Poisson operator matrix that imposes the
+  // corresponding boundary conditions. This type is used as key in other maps,
+  // so it must be hashable. `unordered_map` isn't hashable, but `map` is.
```

Comment:
Is this a Boost thing?  I don't see anything relevant about `std::hash`.  Just be explicit about that in the comment if so.

Reply:
Yes, it's a Boost thing. `boost:hash` exists for `std::map`, but doesn't for `std::unordered_map`. I edited this comment.